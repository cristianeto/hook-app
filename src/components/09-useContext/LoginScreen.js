import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => setUser({ id: "1234", name: "Cristian GuamÃ¡n" })}
      >
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
