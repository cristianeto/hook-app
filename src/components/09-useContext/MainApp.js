import React from "react";

import AppRouter from "./AppRouter";
import { UserContext } from "./UserContext";

const MainApp = () => {
  const user = {
    id: 1234,
    name: "Fernando",
    email: "fernando.herrera@gmail.com",
  };

  return (
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  );
};

export default MainApp;
