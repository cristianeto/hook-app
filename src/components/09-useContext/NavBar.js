import React from "react";
import { Link, NavLink } from "react-router-dom";

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">
        <NavLink to="/" className="navbar-brand">
          UseContext
        </NavLink>

        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink activeClassName="active" className="nav-link" exact to="/">
              Home
            </NavLink>
            <NavLink
              activeClassName="active"
              className="nav-link"
              exact
              to="/about"
            >
              About
            </NavLink>
            <NavLink
              activeClassName="active"
              className="nav-link"
              exact
              to="/login"
            >
              Login
            </NavLink>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
