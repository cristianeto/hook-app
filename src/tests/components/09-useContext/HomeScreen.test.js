import { mount, shallow } from "enzyme";
import React from "react";
import HomeScreen from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Testing in <HomeScreen/> component", () => {
  const user = {
    name: "Cris",
    email: "cris@domain.com",
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test("should show the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
