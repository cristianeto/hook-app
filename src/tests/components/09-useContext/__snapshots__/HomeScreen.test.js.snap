// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <HomeScreen/> component should show the component correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Cris",
   "email": "cris@domain.com"
}
  </pre>
</div>
`;
